src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,93 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T>,
+        MachineProducer<T> {
```

Comment:
```suggestion
public class MachineServiceImpl implements MachineService<Machien> {
```
цей клас не є продюсером, краще не поєднювати декілька поведінок в одному класі а розмежовувати відповідальність за різні задачі. Так код легше підтирмувати. Відповідно до завдання давай створимо TruckProducer, BulldozerProducer... які будуть відповідати за генерацію машин